.gitpod.yml

```diff
@@ -0,0 +1,60 @@
+# Building pandas on init
+# Might delegate this later to prebuild with Q2 improvements on gitpod
+# https://www.gitpod.io/docs/config-start-tasks/#configuring-the-terminal
+# -------------------------------------------------------------------------
+
+# assuming we use dockerhub: name of the docker user, docker image, tag, e.g. https://hub.docker.com/r/pandas/pandas-gitpod/tags
+image:
+  file: pandas-dev/pandas-gitpod:latest
+tasks:
+  - name: Prepare development environment
+    init: |
+      mkdir -p .vscode
+      cp gitpod/settings.json .vscode/settings.json
+      conda activate pandas-dev
+      git pull --unshallow  # need to force this else the prebuild fails
+      git fetch --tags 
+      python setup.py build_ext -j 4
+      python -m pip install -e . --no-build-isolation
+      echo "🛠 Completed rebuilding Pandas!! 🛠 "
+      echo "✨ Pre-build complete! You can close this terminal ✨ "
+
+# --------------------------------------------------------
+# exposing ports for liveserve
+ports:
+  - port: 5500
+    onOpen: notify
+
+# --------------------------------------------------------
+# some useful extensions to have
+vscode:
```

Comment:
Is this common to add for gitpod? I don’t think all of our developers use VSCode so am somewhat hesitant to have a say on what “good” extensions are

Reply:
> I don’t think all of our developers use VSCode

We might rather want to look at contributors that _do_ use VSCode, and see which extensions those typically use, to ensure we can give a good base experience here (if you don't use VSCode, you also won't have expectations about which extension should be enabled, so as long we provide a useful set, those users will be happy I think)

In any case I suppose we want to enable _some_ extensions. For example at least the python extension? Further of course it's a bit the question how far we want to go (eg restructuredtext highlighting can be useful since our docs use that a lot, maybe cython highlighting could be added as well)

Reply:
these seem useful, but I'd remove `autodocstring` as docstrings in pandas are often created via decorators